cpp/src/io/utilities/wrapper_utils.hpp

```diff
@@ -146,3 +146,39 @@ struct rmm_deleter {
 };
 template <typename T>
 using device_ptr = std::unique_ptr<T, rmm_deleter<T>>;
+
+// DOXY
+template <typename T>
```

Comment:
I'd prefer to see this be untyped and just work on raw bytes.

Reply:
hm, in that case the constructor would have to take the number of bytes?

What is the reason for your preference? I like the templated version a lot more because it makes the constructor call much cleaner (no need for sizeof) and also I get the corrent pointer type when getting the raw pointer. With viod*, many calls to data() would require a cast. Overall, I think it's less error prove to keep the type info in the wrapper type.

Reply:
Yeah, you're right.

I was thinking about this as if it were an allocator. One of the biggest mistakes in C++ is making `std::allocator` typed. 

But this isn't an allocator, it's a buffer object. So it makes sense for it to be typed.